src/TabSwipe/widget/TabSwipe.ts

```diff
@@ -0,0 +1,174 @@
+            this.targetWidget.tabSwipeId = this.id;
+            this.targetNode.classList.add(this.swipeClass);
+        }
+    }
+
+    update(_contextObject: mendix.lib.MxObject, callback?: () => void) {
+        if (this.targetWidget) {
+            this.applyIndicatorStyles(this.targetWidget);
+            this.initializeSwipe(this.targetWidget);
+            this.setupEvents(this.targetWidget);
+        }
+
+        if (callback) {
+            callback();
+        }
+    }
+
+    uninitialize() {
+        if (this.swipeHandler) {
+            this.swipeHandler.destroy();
+        }
+
+        return true;
+    }
+
+    private findTargetNode(name: string): HTMLElement {
+        let queryNode = this.domNode.parentNode as Element;
+        let targetNode: HTMLElement | null = null;
+        while (!targetNode) {
+            targetNode = queryNode.querySelector(".mx-name-" + name) as HTMLElement;
+            if (window.document.isEqualNode(queryNode)) { break; }
+            queryNode = queryNode.parentNode as HTMLElement;
+        }
+
+        return targetNode;
+    }
+
+    private checkCompatibility(widget?: TabContainer): boolean {
+        let errorMessage = "";
+        if (!widget) {
+            errorMessage = `" unable to find a target tab with the name ${this.targetName}`;
+        } else if (widget.declaredClass !== this.targetWidgetType) {
+            errorMessage = `target widget ${this.targetName} is not of the type " + ${this.targetWidgetType}`;
+        } else if (!widget._tabList || !widget._tabContent || !widget.showTab || !widget.onShowTab || !widget.onHideTab) { // tslint:disable-line max-line-length
+            errorMessage = "The widget is not compatible with this mendix version.";
+        } else if (widget.tabSwipeId) {
+            errorMessage = `TabContainer '${this.targetName}' is already connected to the widget '` +
+                `${widget.tabSwipeId}. It can only be connected with one widget.`;
+        }
+
+        if (errorMessage) {
+            this.showError(`Tab swipe configuration error: ${errorMessage}`);
+        }
+
+        return !errorMessage;
+    }
+
+    private initializeSwipe(targetWidget: TabContainer) {
+        if (this.swipeHandler) {
+            this.swipeHandler.destroy();
+        }
+        this.swipeHandler = new SwipeHandler({
+            tabContainer: targetWidget,
+            tabContainerContent: targetWidget.domNode.querySelector(this.tabContentClass) as HTMLElement,
+            lazyLoad: this.lazyLoad
+        });
+    }
+
+    private applyIndicatorStyles(targetWidget: TabContainer) {
+        if (this.tabNavStyle === "indicators") {
+            this.targetNode.classList.add("carousel");
+            targetWidget._tabList.classList.remove("nav", "nav-tabs", "mx-tabcontainer-tabs");
+            targetWidget._tabList.classList.add("carousel-indicators");
+        } else {
+            this.targetNode.classList.remove("carousel");
```

Comment:
Remove? this will not change.. so it does not need to be removed? remove `else {/**/}`